GeeksCoreLibrary/Modules/Payments/Services/PayNLService.cs

```diff
@@ -0,0 +1,189 @@
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json.Linq;
+using RestSharp;
+using RestSharp.Authenticators;
+
+namespace GeeksCoreLibrary.Modules.Payments.Services;
+
+/// <inheritdoc cref="IPaymentServiceProviderService" />
+public class PayNlService : PaymentServiceProviderBaseService, IPaymentServiceProviderService, IScopedService
+{
+    private const string BaseUrl = "https://rest.pay.nl/";
+    private readonly ILogger<PaymentServiceProviderBaseService> logger;
+    private readonly IHttpContextAccessor httpContextAccessor;
+    private readonly GclSettings gclSettings;
+    private readonly IShoppingBasketsService shoppingBasketsService;
+
+    public PayNlService(
+        IDatabaseHelpersService databaseHelpersService, 
+        IDatabaseConnection databaseConnection, 
+        ILogger<PaymentServiceProviderBaseService> logger, 
+        IOptions<GclSettings> gclSettings, 
+        IShoppingBasketsService shoppingBasketsService, 
+        IHttpContextAccessor httpContextAccessor = null) : base(databaseHelpersService, databaseConnection, logger, httpContextAccessor)
+    {
+        this.logger = logger;
+        this.shoppingBasketsService = shoppingBasketsService;
+        this.gclSettings = gclSettings.Value;
+        this.httpContextAccessor = httpContextAccessor;
+    }
+
+    public async Task<PaymentRequestResult> HandlePaymentRequestAsync(ICollection<(WiserItemModel Main, List<WiserItemModel> Lines)> conceptOrders, WiserItemModel userDetails,
+        PaymentMethodSettingsModel paymentMethodSettings, string invoiceNumber)
+    {
+        var payNlSettings = (PayNLSettingsModel)paymentMethodSettings.PaymentServiceProvider;
+        var validationResult = ValidatePayNLSettings(payNlSettings);
+        if (!validationResult.Valid)
+        {
+            logger.LogError(validationResult.Message);
+            return new PaymentRequestResult
+            {
+                Successful = false,
+                Action = PaymentRequestActions.Redirect,
+                ActionData = payNlSettings.FailUrl
+            };
+        }
+
+        var totalPrice = await CalculatePriceAsync(conceptOrders);
+
+        // Build and execute payment request.
+        var restClient = CreateRestClient(payNlSettings);
+        var restRequest = CreateTransactionStartRequest(totalPrice, payNlSettings, invoiceNumber);
+        var restResponse = await restClient.ExecuteAsync(restRequest);
+        var responseJson = JObject.Parse(restResponse.Content);
+        var responseSuccessful = restResponse.StatusCode == HttpStatusCode.Created;
+        
+        return new PaymentRequestResult
+        {
+            Successful = responseSuccessful,
+            Action = PaymentRequestActions.Redirect,
+            ActionData = (responseSuccessful) ? responseJson["paymentUrl"]?.ToString() : payNlSettings.FailUrl
+        };
+    }
+
+    public async Task<StatusUpdateResult> ProcessStatusUpdateAsync(OrderProcessSettingsModel orderProcessSettings,
```

Comment:
Graag `/// <inheritdoc />` toevoegen boven de functie, zodat de uitleg die in de interface staat wordt overgenomen naar de implementatie.